crates/tauri-runtime/src/lib.rs

```diff
@@ -437,6 +437,10 @@ pub trait Runtime<T: UserEvent>: Debug + Sized + 'static {
   #[cfg(desktop)]
   fn run_iteration<F: FnMut(RunEvent<T>) + 'static>(&mut self, callback: F);
 
+  /// Equivalent to [`Runtime::run`] but returns the exit code instead of exiting the process.
+  #[cfg(desktop)]
+  fn run_return<F: FnMut(RunEvent<T>) + 'static>(&mut self, callback: F) -> i32;
```

Comment:
Adding a new, non-default function to a public trait is not semver compatible. Is it okay to bump this to `3.0` or should we try and find a semver-compatible way?

Reply:
Unfortunately, `tauri_runtime`'s version leaks into the public API of `tauri` through e.g. this function: https://github.com/thomaseizinger/tauri/blob/9be493e2cf56e951e5ba1e5956ba46a18854bf7f/crates/tauri/src/app.rs#L144

Therefore, bumping `tauri-runtime` to major would also mean bumping `tauri` to major if we are being strict about semver.

Reply:
I decided to give it a default impl that panics.

Reply:
Only just read that we are not following semver with this crate.